File: Programowanie_Obiektowe/K1-Probne/court.py
from datetime import datetime


class Court:
    __width: float
    __length: float
    __address: str
    __year_built: int

    def __init__(self, width: float = 68.0, length: float = 100.0,
                 address: str = '', year_built: int = 0) -> None:
        if 45 <= width <= 90 <= length <= 120:
            self.__width = width
            self.__length = length
        else:
            self.__width = 68.0
            self.__length = 100.0
        self.__address = address
        self.__year_built = year_built

    @property
    def width(self) -> float:
        return self.__width

    @property
    def length(self) -> float:
        return self.__length

    @property
    def address(self) -> str:
        return self.__address

    @property
    def year_built(self) -> int:
        return self.__year_built

    @width.setter
    def width(self, value: float) -> None:
        if 90 <= value <= 120:
            self.__width = value
        else:
            print('Nieprawidłowa wielkość')

    @length.setter
    def length(self, value: float) -> None:
        if 90 <= value <= 120:
            self.__length = value
        else:
            print('Nieprawidłowa wielkość')

    @address.setter
    def address(self, value: str) -> None:
        self.__address = value

    @year_built.setter
    def year_built(self, value: int) -> None:
        self.__year_built = value

    def area(self) -> float:
        return self.__length * self.__width

    @staticmethod
    def validate(court: 'Court') -> None:
        x = int(datetime.now().year)
        if court.__year_built <= 0 or court.__year_built > x:
            court.__year_built = x

    def __str__(self):
        return f'Boisko wybudowane w roku' \
               f' {self.__year_built},'f' o ' \
               f'długości {self.__length} metrów i ' \
               f'szerokości {self.__width} metrów.\nPole' \
               f'powierzchni: {self.area()} mkw.\nAdres:' \
               f' {self.__address} '

    def __eq__(self, other: 'Court') -> bool:
        return self.area() == other.area

    def __ne__(self, other: 'Court') -> bool:
        return self.area() != other.area

    def __int__(self):
        return 5

File: Programowanie_Obiektowe/K2-Probne/Zestaw_1/Court.java
package pl.uwm.wmii.kaz.lemmgator;

import java.time.LocalDate;
import java.util.Objects;

public class Court {
    private double width;
    private double length;
    private String address;
    private int year_built;

    public Court(double width, double length, String address, int year_built) {
        if (width < 45 || width > 90 || length < 90 || length > 120) {
            this.width = 68;
            this.length = 100;
        } else {
            this.width = width;
            this.length = length;
        }
        this.address = address;
        this.year_built = year_built;
    }

    public Court(String address, int year_built) {
        this(68, 100, address, year_built);
    }

    public double getWidth() {
        return width;
    }

    public void setWidth(double width) {
        if (width < 45 || width > 90) {
            this.width = 68;
        } else {
            this.width = width;
        }
    }

    public double getLength() {
        return length;
    }

    public void setLength(double length) {
        if (length < 90 || length > 120) {
            this.length = 100;
        } else {
            this.length = length;
        }
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public int getYear_built() {
        return year_built;
    }

    public void setYear_built(int year_built) {
        this.year_built = year_built;
    }

    public static void validate(Court court) {
        int now = LocalDate.now().getYear();
        if (court.year_built < 0 || court.year_built > now) {
            court.year_built = now;
        }
    }

    public double area() {
        return getWidth() * getLength();
    }

    @Override
    public String toString() {
        return "Boisko wybudowane w roku " + this.year_built + ", o dlugosci " + this.length + " metrow i szerokosci " + this.width + " metrow.\n" +
                "Pole powierzchni: " + area() + " mkw.\n" +
                "Adres: " + this.address + ".";
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null) return false;
        if (!(o instanceof Court court)) return false;
        return Double.compare(court.width, width) == 0 && Double.compare(court.length, length) == 0;
    }
}